### PR TITLE
[WIP] Update Yaml stages guidance

### PR DESCRIPTION
**Pull request by riarenas**
This should be the last PR to close https://github.com/dotnet/arcade/issues/3389

Still need to add:

* Call out explicitly the pattern for custom publishing: Call PushToAzureDevOpsArtifacts to generate the manifests and create the correct build artifacts, and then copy the generated manifest over. We have an example in the doc already, but I want to expand that this is for every time you are thinking of publishing.

  * Explanation for the buildArtifacts:
  * PackageArtifacts
  * BlobArtifacts
  * PdbArtifacts

### CODE REVIEW

File: Documentation/CorePackages/YamlStagesPublishing.md [hunk 1]
```diff
@@ -30,6 +30,25 @@ In order to use this new publishing mechanism, the easiest way to start is by ma
 
 1. Update the repo's arcade version to `1.0.0-beta.19360.8` or newer.
 
+1. Add an artifact category for assets produced by the build:
+
+    * Most repositories will use `.NETCore` as the category unless assets should be published to a feed
+    other than `dotnet-core`.
+    See [Advanced Scenarios](##overriding-the-publishing-feed-used-for-builds-in-the-dev-channel) for instructions on how to publish to a different feed.
+
+    * `_DotNetValidationArtifactsCategory` for publishing to the Validation channel (Such as for testing the changes in these onboarding steps)
```
Comment (JohnTortugo) [inline Documentation/CorePackages/YamlStagesPublishing.md]: As talked on Teams I'll line this value so we won't need the variable anymore.

File: Documentation/CorePackages/YamlStagesPublishing.md [hunk 2]
```diff
@@ -30,6 +30,25 @@ In order to use this new publishing mechanism, the easiest way to start is by ma
...
+    * `_DotNetArtifactsCategory` for publishing to the Dev channel.
+
+      ```YAML
+      variables:
+      ...
+      - name: _DotNetValidationArtifactsCategory
```
Comment (JohnTortugo) [inline Documentation/CorePackages/YamlStagesPublishing.md]: No need for this anymore.

File: Documentation/CorePackages/YamlStagesPublishing.md [hunk 3]
```diff
@@ -179,9 +213,7 @@ for the category to the TargetStaticFeed PropertyGroup in
     </PropertyGroup>
     ```
 
-1. Add a variable to your pipeline's YAML and set the value as your new category.
-    * `_DotNetValidationArtifactsCategory` for publishing to the Validation channel (Such as for testing the changes in these onboarding steps)
-    * `_DotNetArtifactsCategory` for publishing to the Dev channel.
+1. Set your new category as the value for the artifact category variables
```
Comment (JohnTortugo) [inline Documentation/CorePackages/YamlStagesPublishing.md]: Won't be possible to override the feed for the validation channel anymore. The value of `_DotNetValidationArtifactsCategory` will be hardcoded.

File: Documentation/CorePackages/YamlStagesPublishing.md [hunk 4]
```diff
@@ -69,6 +88,21 @@ In order to use this new publishing mechanism, the easiest way to start is by ma
...
+    | Name                            | Value                                                                               |
+    | --------------------------------| ----------------------------------------------------------------------------------- |
+    | /p:DotNetPublishBlobFeedUrl     | https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json                     |
+    | /p:DotNetPublishBlobFeedKey     | `$(dotnetfeed-storage-access-key-1)` variable from Dotnet-Blob-Feed variable group  |
+    | /p:DotNetPublishToBlobFeed      | true                                                                                |
+    | /p:DotNetPublishUsingPipelines  | true                                                                                |
+    | /p:DotNetArtifactsCategory      | `$(_DotNetArtifactsCategory)` variable                                              |
+
+    For an example, see how the Arcade repo passes these properties in its
+    [azure-pipelines.yml](https://github.com/dotnet/arcade/blob/2cb8b86c1ca7ff77304f76fe7041135209ab6932/azure-pipelines.yml#L74)
```
Comment (jaredpar) [inline Documentation/CorePackages/YamlStagesPublishing.md]: Is there a list of properties that repositories may be passing that are invalidated by this move? For instance `/p:PublishToSymbolServer`

Comment (riarenas) [reply Documentation/CorePackages/YamlStagesPublishing.md]: We haven't taken an inventory of the properties that are now obsolete. That will be done as part of https://github.com/dotnet/arcade/issues/3597

File: Documentation/CorePackages/YamlStagesPublishing.md (hunk 1 above)
Comment (riarenas) [reply Documentation/CorePackages/YamlStagesPublishing.md]: That work is still pending, so I'm not changing it here.

**Comment by riarenas** [comment]
@johntortugo please take another look. Added the newer properties that have been added.

File: Documentation/CorePackages/YamlStagesPublishing.md (hunk 3 above)
Comment (riarenas) [reply Documentation/CorePackages/YamlStagesPublishing.md]: That change didn't make it in yet.

**Pull request merged**